pkg/identity/identity_test.go

```diff
@@ -237,6 +237,93 @@ func TestLookupReservedIdentityByLabels(t *testing.T) {
 	}
 }
 
+func TestIPIdentityPair_PrefixString(t *testing.T) {
+	ipv6Mask := make(net.IPMask, net.IPv6len)
+	for i := range ipv6Mask {
+		ipv6Mask[i] = 255
+	}
+
+	tests := []struct {
+		name     string
+		expected string
+		pair     *IPIdentityPair
+	}{
+		{
+			name:     "IPv4 with mask",
+			expected: "10.1.128.15/32",
+			pair: &IPIdentityPair{
+				IP:           net.ParseIP("10.1.128.15"),
+				Mask:         net.IPv4Mask(255, 255, 255, 255),
+				HostIP:       net.ParseIP("10.1.128.15"),
+				ID:           1,
+				Key:          3,
+				Metadata:     "metadata",
+				K8sNamespace: "kube-system",
+				K8sPodName:   "host",
+				NamedPorts: []NamedPort{
+					{Name: "port", Port: 8080, Protocol: "TCP"},
+				},
+			},
+		},
+		{
+			name:     "IPv4 without mask",
+			expected: "10.1.128.15",
+			pair: &IPIdentityPair{
+				IP:           net.ParseIP("10.1.128.15"),
+				HostIP:       net.ParseIP("10.1.128.15"),
+				ID:           1,
+				Key:          3,
+				Metadata:     "metadata",
+				K8sNamespace: "kube-system",
+				K8sPodName:   "host",
+				NamedPorts: []NamedPort{
+					{Name: "port", Port: 8080, Protocol: "TCP"},
+				},
+			},
+		},
+		{
+			name:     "IPv6 with mask",
+			expected: "10.1.128.15/128",
+			pair: &IPIdentityPair{
+				IP:           net.ParseIP("::ffff:a01:800f"),
+				Mask:         ipv6Mask,
+				HostIP:       net.ParseIP("::ffff:a01:800f"),
```

Comment:
Shouldn't the expected IP here be in v6 form?

Reply:
I believe this is the existing behavior due to how golang's parse IP handles IPv4 addresses that are encoded as IPv6 notation. 

I will add some test coverage for pure IPv6 addresses like local `fd12:3456:789a:1::1`